phases/01-math-foundations/05-chain-rule-and-autodiff/code/autodiff.py

```
@@ -168,6 +188,138 @@ def demo_neuron():
         print("  PASSED (relu inactive, all grads zero)\n")
 
 
+class Neuron:
+    def __init__(self, n_inputs):
+        self.w = [Value(random.uniform(-1, 1)) for _ in range(n_inputs)]
+        self.b = Value(0.0)
+
+    def __call__(self, x):
+        act = sum((wi * xi for wi, xi in zip(self.w, x)), self.b)
```

Ruff (0.15.6)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)